src/index.js

```diff
@@ -0,0 +1,194 @@
+
+/**
+ * A higher-order-component for handling onClickOutside for React components.
+ */
+const registeredComponents = [];
+const handlers = [];
+
+/**
+ * This function generates the HOC function that you'll use
+ * in order to impart onOutsideClick listening to an
+ * arbitrary component. It gets called at the end of the
+ * bootstrapping code to yield an instance of the
+ * onClickOutsideHOC function defined inside setupHOC().
+ */
+export default function onClickOutsideHOC(WrappedComponent, config) {
+  return class onClickOutside extends Component {
+    static displayName = `OnClickOutside(${ WrappedComponent.displayName || WrappedComponent.name || 'Component' })`
+
+    static defaultProps = {
+      eventTypes: ['mousedown', 'touchstart'],
+      excludeScrollbar: (config && config.excludeScrollbar) || false,
+      outsideClickIgnoreClass: 'ignore-react-onclickoutside',
+      preventDefault: false,
+      stopPropagation: false,
+    }
+
+    static getClass = () => WrappedComponent.getClass ? WrappedComponent.getClass() : WrappedComponent
+
+    /**
+     * Access the WrappedComponent's instance.
+     */
+    getInstance() {
+      return WrappedComponent.prototype.isReactComponent ? this.instanceRef : this;
+    }
+
+    // this is given meaning in componentDidMount
+    __outsideClickHandler() {}
+
+    /**
+     * Add click listeners to the current document,
+     * linked to this component's state.
+     */
+    componentDidMount() {
+      // If we are in an environment without a DOM such
+      // as shallow rendering or snapshots then we exit
+      // early to prevent any unhandled errors being thrown.
+      if (typeof document === 'undefined' || !document.createElement){
+        return;
+      }
+
+      const instance = this.getInstance();
+      let clickOutsideHandler;
+
+      if(config && typeof config.handleClickOutside === 'function') {
+        clickOutsideHandler = config.handleClickOutside(instance);
+        if(typeof clickOutsideHandler !== 'function') {
+          throw new Error('WrappedComponent lacks a function for processing outside click events specified by the handleClickOutside config option.');
+        }
+      } else if(typeof instance.handleClickOutside === 'function') {
+        if (Component.prototype.isPrototypeOf(instance)) {
+          clickOutsideHandler = instance.handleClickOutside.bind(instance);
+        } else {
+          clickOutsideHandler = instance.handleClickOutside;
+        }
+      } else if(typeof instance.props.handleClickOutside === 'function') {
+        clickOutsideHandler = instance.props.handleClickOutside;
+      } else {
+        throw new Error('WrappedComponent lacks a handleClickOutside(event) function for processing outside click events.');
+      }
+
+      // TODO: try to get rid of this, could be done with function ref, might be problematic for SFC though, they do not expose refs
+      const componentNode = this.instanceRef;
```

Comment:
on the master:
L53: `var instance = this.getInstance();`
L74: `var componentNode = ReactDOM.findDOMNode(instance);`

I got rid of ReactDOM call here, but it might have been the bad idea. Later in the render we have such block:
```javascript
      if (WrappedComponent.prototype.isReactComponent) {
        props.ref = this.getRef;
      } else {
        props.wrappedRef = this.getRef;
      }
```
and now it will break, as if `WrappedComponent` is a class one we will get access to its instance and not to the underlaying DOM node, we could skip the if and just expose `wrappedRef` to both class and functional components, so they can attach ref to whatever node they wish, but thats certainly a breaking change and because of the new API aint sure if we should include this.

Please decide what should be done here, but the most reasonable thing seems to be reverting the change and keeping `ReactDOM.findDOMNode`